ts/src/toobit.ts

```diff
@@ -0,0 +1,2894 @@
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+     */
+    async createOrder (symbol: string, type: OrderType, side: OrderSide, amount: number, price: Num = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        let request = {};
+        let response = undefined;
+        if (market['spot']) {
+            [ request, params ] = this.createOrderRequest (symbol, type, side, amount, price, params);
+            response = await this.privatePostApiV1SpotOrder (this.extend (request, params));
+        } else {
+            [ request, params ] = this.createContractOrderRequest (symbol, type, side, amount, price, params);
+            response = await this.privatePostApiV1FuturesOrder (this.extend (request, params));
+        }
+        //
+        //     {
+        //         "symbol": "ETHUSDT",
+        //         "price": "0",
+        //         "origQty": "0.001",
+        //         "orderId": "2024837825254460160",
+        //         "clientOrderId": "1756115478113679",
+        //         "executedQty": "0",
+        //         "status": "PENDING_NEW",
+        //         "timeInForce": "GTC",
+        //         "type": "MARKET",
+        //         "side": "SELL"
+        //         "accountId": "1783404067076253952",    // only in spot
+        //         "symbolName": "ETHUSDT",               // only in spot
+        //         "transactTime": "1756115478604",       // only in spot
+        //         "time": "1668418485058",               // only in contract
+        //         "updateTime": "1668418485058",         // only in contract
+        //         "leverage": "2",                       // only in contract
+        //         "avgPrice": "0",                       // only in contract
+        //         "marginLocked": "9.5",                 // only in contract
+        //         "priceType": "INPUT"                   // only in contract
+        //     }
+        //
+        return this.parseOrder (response, market);
+    }
+
+    createOrderRequest (symbol: string, type: OrderType, side: OrderSide, amount: number, price: Num = undefined, params = {}) {
+        const market = this.market (symbol);
+        const id = market['id'];
+        const request: Dict = {
+            'symbol': id,
+            'side': side.toUpperCase (),
+            'price': this.priceToPrecision (symbol, price),
+        };
+        let cost: Str = undefined;
+        [ cost, params ] = this.handleParamString (params, 'cost');
+        if (type === 'market') {
+            if (cost === undefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder() requires params["cost"] for market orders');
+            } else {
+                request['quantity'] = this.amountToPrecision (symbol, cost);
```

Comment:
shouldn't we use `costToPrecision` here?

Also is this true for both buy/sell orders?